### PR TITLE
Issue #9907: set up Java16 build

### PR DESCRIPTION
**Pull request by pbludov**
Solves #9907 

* Replace Java15 build with Java16 build in Cirrus CI
* ~~Add javac16 stage to Drone CI~~

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -86,6 +86,11 @@ steps:
   commands:
   - ./.ci/travis/travis.sh javac15
 
+- name: javac16
+  image: maven:3.6.3-adoptopenjdk-16
```
Comment (nmancus1) [inline .drone.yml:90]: Just curious, why this image and not `maven:3.8.1-openjdk-16` ?

**Review by nmancus1 — APPROVED** [review]

**Comment by nmancus1** [comment]
I am good once CI passes.

File: .drone.yml (hunk 1 above)
Comment (pbludov) [reply .drone.yml]: Good question. Changed to 3.8.1.
If this version turns out to be incompatible, I will revert it back and create a separate issue.

**Comment by pbludov** [comment]
Related powermock issue https://github.com/powermock/powermock/issues/1094

**Comment by pbludov** [comment]
As a workaround, the option `--illegal-access=warn` can be used for tests.

**Comment by rnveach** [comment]
As pointed out at https://github.com/checkstyle/checkstyle/pull/9902#issuecomment-825662057 , we should just close this PR and finish reviewing and merging the other.

**Review by nmancus1 — APPROVED** [review]

**Review by romani — APPROVED** [review]

**Pull request merged**